warning during the Debian build

Hello armando, here a bunch of warning observed during the build

```
kages/numpy/core/include -I/<<PKGBUILDDIR>>/src -I/usr/include/python3.6dm -c /<<PKGBUILDDIR>>/src/fisx_simpleini.cpp -o build/temp.linux-amd64-3.6-pydebug/<<PKGBUILDDIR>>/src/fisx_simpleini.o
/<<PKGBUILDDIR>>/src/fisx_simpleini.cpp: In member function 'void fisx::SimpleIni::readFileName(std::__cxx11::string)':
/<<PKGBUILDDIR>>/src/fisx_simpleini.cpp:94:17: warning: 'p0' may be used uninitialized in this function [-Wmaybe-uninitialized]
                 if (p0 != 0)
                 ^~
x86_64-linux-gnu-gcc -pthread -g -Og -Wall -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/lib/python3/dist-packages/numpy/core/include -I/<<PKGBUILDDIR>>/src -I/usr/include/python3.6dm -c /<<PKGBUILDDIR>>/src/fisx_xrfconfig.cpp -o build/temp.linux-amd64-3.6-pydebug/<<PKGBUILDDIR>>/src/fisx_xrfconfig.o
/<<PKGBUILDDIR>>/src/fisx_xrfconfig.cpp: In member function 'void fisx::XRFConfig::readConfigurationFromFile(const string&)':
/<<PKGBUILDDIR>>/src/fisx_xrfconfig.cpp:68:10: warning: variable 'fisxFile' set but not used [-Wunused-but-set-variable]
     bool fisxFile;
          ^~~~~~~~
x86_64-linux-gnu-gcc -pthread -g -Og -Wall -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/lib/python3/dist-packages/numpy/core/include -I/<<PKGBUILDDIR>>/src -I/usr/include/python3.6dm -c /<<PKGBUILDDIR>>/src/fisx_version.cpp -o build/temp.linux-amd64-3.6-pydebug/<<PKGBUILDDIR>>/src/fisx_version.o
x86_64-linux-gnu-gcc -pthread -g -Og -Wall -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/lib/python3/dist-packages/numpy/core/include -I/<<PKGBUILDDIR>>/src -I/usr/include/python3.6dm -c /<<PKGBUILDDIR>>/src/fisx_elements.cpp -o build/temp.linux-amd64-3.6-pydebug/<<PKGBUILDDIR>>/src/fisx_elements.o
/<<PKGBUILDDIR>>/src/fisx_elements.cpp: In member function 'void fisx::Elements::setMassAttenuationCoefficients(const string&, const std::vector<double>&, const std::vector<double>&, const std::vector<double>&, const std::vector<double>&, const std::vector<double>&)':
/<<PKGBUILDDIR>>/src/fisx_elements.cpp:813:33: warning: comparison of integer expressions of different signedness: 'std::vector<double>::size_type' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
                 if ((i - j - 1) == extractedEdgeEnergies[shell].second)
                     ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/<<PKGBUILDDIR>>/src/fisx_elements.cpp: In member function 'std::map<std::__cxx11::basic_string<char>, std::map<std::__cxx11::basic_string<char>, double> > fisx::Elements::getEscape(const std::map<std::__cxx11::basic_string<char>, double>&, const double&, const double&, const double&, const int&, const double&, const double&) const':
/<<PKGBUILDDIR>>/src/fisx_elements.cpp:1954:16: warning: variable 'factor' set but not used [-Wunused-but-set-variable]
         double factor;
                ^~~~~~
/<<PKGBUILDDIR>>/src/fisx_elements.cpp: In member function 'std::map<std::__cxx11::basic_string<char>, double> fisx::Elements::parseFormula(const string&) const':
/<<PKGBUILDDIR>>/src/fisx_elements.cpp:1708:13: warning: 'parsingKey' may be used uninitialized in this function [-Wmaybe-uninitialized]
             if (parsingKey)
             ^~
/<<PKGBUILDDIR>>/src/fisx_elements.cpp: In member function 'std::map<std::__cxx11::basic_string<char>, std::map<std::__cxx11::basic_string<char>, double> > fisx::Elements::getEscape(const std::map<std::__cxx11::basic_string<char>, double>&, const double&, const double&, const double&, const int&, const double&, const double&) const':
/<<PKGBUILDDIR>>/src/fisx_elements.cpp:1979:26: warning: 'intrinsicEfficiency' may be used uninitialized in this function [-Wmaybe-uninitialized]
                     rate /= intrinsicEfficiency;
                     ~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/fisx_detector.cpp
+++ b/src/fisx_detector.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -130,22 +130,29 @@ std::map<std::string, std::map<std::string, double> > Detector::getEscape(const 
                                                             const std::string & label,
                                                             const int & update)
 {
-    if (update != 0)
-        this->escapePeakCache.clear();
-    if (false) //(label.size())
+    // To implement a cache at the detector level does not work because the same
+    // detector can use different elementsLibrary instances and to use different
+    // labels to identify them does not seem a good idea.
+    // It is preferable to fill the cache at the elements library level
+    if (label.size())
     {
-        if (this->escapePeakCache.find(label) == this->escapePeakCache.end())
+        if (update != 0)
+            this->escapePeakCache.clear();
+        if (this->escapePeakCache.find(label) != this->escapePeakCache.end())
         {
-            // calculate it
-            this->escapePeakCache[label] = elementsLibrary.getEscape(this->getComposition(elementsLibrary), \
-                                                                     energy, \
-                                                                     this->escapePeakEnergyThreshold, \
-                                                                     this->escapePeakIntensityThreshold, \
-                                                                     this->escapePeakNThreshold, \
-                                                                     this->escapePeakAlphaIn,
-                                                                     0);
+            if (this->escapePeakCache[label].find(energy) != this->escapePeakCache[label].end())
+            {
+                return this->escapePeakCache[label][energy];
+            }
         }
-        return this->escapePeakCache[label];
+        this->escapePeakCache[label][energy] = elementsLibrary.getEscape(this->getComposition(elementsLibrary), \
+                                                                 energy, \
+                                                                 this->escapePeakEnergyThreshold, \
+                                                                 this->escapePeakIntensityThreshold, \
+                                                                 this->escapePeakNThreshold, \
+                                                                 this->escapePeakAlphaIn,
+                                                                 0);
+        return this->escapePeakCache[label][energy];
     }
     else
     {

--- a/src/fisx_detector.h
+++ b/src/fisx_detector.h
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -104,12 +104,13 @@ public:
     Returns escape peak energy and rate per detected photon of given energy.
 
     The optional arguments label and update serve for caching purposes.
+    To make use of the cache, the label must not be empty (it should be used to identify the elementsLibrary)
+    and update must be set to 0 because if it is set to 1 it will systematically empty the cascade cache.
     */
     std::map<std::string, std::map<std::string, double> > getEscape(const double & energy, \
                                                             const Elements & elementsLibrary, \
                                                             const std::string & label = "", \
                                                             const int & update = 1);
-
     void setMinimumEscapePeakEnergy(const double & energy);
     void setMinimumEscapePeakIntensity(const double & intensity);
     void setMaximumNumberOfEscapePeaks(const int & nPeaks);
@@ -122,7 +123,7 @@ private:
     double escapePeakIntensityThreshold;
     int escapePeakNThreshold;
     double escapePeakAlphaIn;
-    std::map< std::string, std::map<std::string, std::map<std::string, double> > > escapePeakCache;
+    std::map< std::string, std::map< double, std::map<std::string, std::map<std::string, double> > > >escapePeakCache;
     // TODO: Calibration, fano, noise, and so on.
 };
 

--- a/src/fisx_element.cpp
+++ b/src/fisx_element.cpp
@@ -533,6 +533,7 @@ std::map<std::string, std::pair<double, int> > \
         }
     }
 
+    /*
     if (false)
     {
         for (result_it = result.begin(); result_it != result.end(); ++result_it)
@@ -542,6 +543,7 @@ std::map<std::string, std::pair<double, int> > \
             std::cout << this->name << " Found shell " << key << " at " << energy << std::endl;
         }
     }
+    */
     return result;
 }
 
@@ -1280,10 +1282,9 @@ Element::getXRayLinesFromVacancyDistribution(const std::map<std::string, double>
                 if (cacheKey == cascadeCache.end())
                 {
                     std::cout << this->name << " Error processing vacancy on shell " << c_it->first << std::endl;
-                    throw std::runtime_error("Vacancy on a shell not present in the cache!");
+                    throw std::runtime_error("Vacancy in a shell not present in the cache!");
                     // the above is not good, there can be no emission following a vacancy (rate too low)
                     // it seems to be triggered by As L1 vacancies
-                    continue;
                 }
                 for(std::map<std::string, std::map<std::string, double> >::const_iterator rayIt = \
                     cacheKey->second.begin(); rayIt != cacheKey->second.end(); ++rayIt)

--- a/src/fisx_element.h
+++ b/src/fisx_element.h
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -370,7 +370,7 @@ private:
     std::map<std::string, std::vector<double> > muPartialPhotoelectricValue;
 
     // A cache for storing calculations
-    static const int cacheMaximumSize = 10000;
+    static const unsigned int cacheMaximumSize = 10000;
     bool calculationCacheEnabledFlag;
     std::map< double, std::map< std::string, double> > muCache;
     std::map< double, std::map<std::string, std::map<std::string, double> > > excitationFactorsCache;

--- a/src/fisx_elements.cpp
+++ b/src/fisx_elements.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2017 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -810,7 +810,7 @@ void Elements::setMassAttenuationCoefficients(const std::string & name,
             shell = shellList[shellIndex];
             if (extractedEdgeEnergies.find(shell) != extractedEdgeEnergies.end())
             {
-                if ((i - j - 1) == extractedEdgeEnergies[shell].second)
+                if ((i - j - 1) == static_cast<unsigned int>(extractedEdgeEnergies[shell].second))
                 {
                     // We are just above an edge according to the input mass attenuation coefficients
                     // We have to check if that energy is compatible with the binding energy.
@@ -1918,14 +1918,6 @@ std::map<std::string,std::map<std::string, double> > Elements::getEscape( \
             // std::cout << "USING CACHE" <<  energy << std::endl;
             return it->second;
         }
-        else
-        {
-            // std::cout << "ENERGY NOT THERE" << energy << std::endl;
-        }
-    }
-    else
-    {
-        // std::cout << "NOT COMPATIBLE" << std::endl;
     }
 
     if (alphaIn == 90.)
@@ -1951,15 +1943,15 @@ std::map<std::string,std::map<std::string, double> > Elements::getEscape( \
         std::map<std::string, std::map<std::string, double> >::const_iterator it;
         std::map<std::string, double>::const_iterator mapIt;
         // factor uses mu["shell photoelectric"] / mu["total photoelectric"]
-        double factor;
+        // double factor;
         // rate uses mu["shell photoelectric"]
         // It is the product factor * mu["total photoelectric"]
         double rate;
         double fluorescentEnergy;
         for ( it = excitationFactors.begin(); it != excitationFactors.end(); ++it)
         {
-            mapIt = it->second.find("factor");
-            factor = mapIt->second;
+            // mapIt = it->second.find("factor");
+            // factor = mapIt->second;
             mapIt = it->second.find("rate");
             rate = mapIt->second;
             mapIt = it->second.find("energy");

--- a/src/fisx_elements.cpp
+++ b/src/fisx_elements.cpp
@@ -1550,6 +1550,7 @@ std::map<std::string, double> Elements::parseFormula(const std::string & formula
     }
 
     // look for parenthesis
+    parsingKey = false;
     spacesPresent = false;
     for(i=0; i < formula.size(); i++)
     {
@@ -1934,7 +1935,16 @@ std::map<std::string,std::map<std::string, double> > Elements::getEscape( \
     result.clear();
 
     if (thickness > 0.0)
+    {
         intrinsicEfficiency = (1.0 - std::exp(-muIncident * thickness / sinAlphaIn));
+    }
+    else
+    {
+        // this is just to get rid of a not very clever non-initialized warning because
+        // if thickness is not bigger than 0.0 intrinsicEfficiency will not be used in
+        // any case.
+        intrinsicEfficiency = 1.0;
+    }
     for (c_it = composition.begin(); c_it != composition.end(); c_it++)
     {
         element = c_it->first;

--- a/src/fisx_math.cpp
+++ b/src/fisx_math.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -315,7 +315,7 @@ double Math::deBoerL0(const double & mu1, const double & mu2, const double & muj
 double Math::deBoerX(const double & p, const double & q, const double & d1, const double & d2, \
                      const double & mu1j, const double & mu2j, const double & mubj_dt)
 {
-     if(false)
+     if ((false))
      {
         double result;
         result =  Math::deBoerV(p, q, d1, d2, mu1j, mu2j, mubj_dt) - \

--- a/src/fisx_multilayer.cpp
+++ b/src/fisx_multilayer.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2017 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -519,7 +519,8 @@ std::map<std::string, std::map<int, std::map<std::string, std::map<std::string, 
                     //std::cout << c_it->first << "primary = " << result[c_it->first]["primary"] << std::endl;
                     //std::cout << c_it->first << "energy = " << result[c_it->first]["energy"] << std::endl;
                     //std::cout << c_it->first << "mu_1_i = " << result[c_it->first]["mu_1_i"] << std::endl;
-                    if (false && (c_it->first == "KL2") && (iLayer == 0))
+                    /*
+                    if ((c_it->first == "KL2") && (iLayer == 0))
                     {
                         std::cout << c_it->first << "efficiency = " << result[c_it->first]["efficiency"] << std::endl;
                         std::cout << c_it->first << "primary = " << result[c_it->first]["primary"] << std::endl;
@@ -532,6 +533,7 @@ std::map<std::string, std::map<int, std::map<std::string, std::map<std::string, 
                         std::cout << c_it->first << "mu_1_lambda/sinALphain = " << mu_1_lambda / sinAlphaIn<< std::endl;
                         std::cout << c_it->first << "mu_1_i/sinALphaOut = " << mu_1_i / sinAlphaOut<< std::endl;
                     }
+                    */
                 }
 
                 if (secondary > 0)

--- a/src/fisx_simpleini.cpp
+++ b/src/fisx_simpleini.cpp
@@ -66,6 +66,11 @@ void SimpleIni::readFileName(std::string fileName)
     std::ifstream fileInstance(fileName.c_str(), std::ios::in | std::ios::binary);
     std::string msg;
 
+    if (!fileInstance.good())
+    {
+        msg = "File <" + fileName + "> mot accessible";
+        throw std::ios_base::failure(msg);;
+    }
     this->sections.clear();
     this->sectionPositions.clear();
     position = 0;

--- a/src/fisx_simpleini.cpp
+++ b/src/fisx_simpleini.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #

--- a/src/fisx_simpleini.cpp
+++ b/src/fisx_simpleini.cpp
@@ -73,6 +73,7 @@ void SimpleIni::readFileName(std::string fileName)
     }
     this->sections.clear();
     this->sectionPositions.clear();
+    p0 = 0;
     position = 0;
     numberOfLines = -1;
     while (std::getline(fileInstance, line))

--- a/src/fisx_xrf.cpp
+++ b/src/fisx_xrf.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -192,7 +192,6 @@ std::map<std::string, std::map<int, std::map<std::string, std::map<std::string, 
     std::vector<std::string> elementList;
     std::vector<std::string> familyList;
     std::vector<int> layerList;
-    std::vector<std::string>::size_type i;
     std::string tmpString;
     std::vector<std::string> tmpStringVector;
 

--- a/src/fisx_xrfconfig.cpp
+++ b/src/fisx_xrfconfig.cpp
@@ -2,7 +2,7 @@
 #
 # The fisx library for X-Ray Fluorescence
 #
-# Copyright (c) 2014-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2014-2018 European Synchrotron Radiation Facility
 #
 # This file is part of the fisx X-ray developed by V.A. Sole
 #
@@ -65,12 +65,14 @@ void XRFConfig::readConfigurationFromFile(const std::string & fileName)
     std::vector<std::string> splitString;
     Material material;
     Layer layer;
-    bool fisxFile;
     long counter;
     bool multilayerSample;
     double value;
 
+
     // find out if it is a fix or a PyMca configuration file
+    /* This not needed since a fisx file format is not defined yet
+    bool fisxFile;
     sectionContents.clear();
     sectionContents = iniFile.readSection("fisx", false);
     fisxFile = true;
@@ -78,11 +80,10 @@ void XRFConfig::readConfigurationFromFile(const std::string & fileName)
     {
         fisxFile = false;
     }
-    /*
     if (fisxFile)
     {
         std::cout << "Not implemented" << std::endl;
-        return;
+        throw std::invalid_argument("Only PyMca configuration files supported.");
     }
     */
 
@@ -98,7 +99,7 @@ void XRFConfig::readConfigurationFromFile(const std::string & fileName)
         sectionContents = iniFile.readSection("result.config", false);
         if(!sectionContents.size())
         {
-            throw std::invalid_argument("File not recognized as a fisx or PyMca configuration file.");
+            throw std::invalid_argument("File not recognized as a PyMca configuration file.");
         }
         std::cout << "fit result file" << std::endl;
     }

